Experius patch registration address

lewisvoncken opened the pull request:
<!--- Provide a general summary of the Pull Request in the Title above -->

### Description
<!--- Provide a description of the changes proposed in the pull request -->
Add configuration setting to Show Address Fields on the Registration page


### Manual testing scenarios
<!--- Provide a set of unambiguous steps to test the proposed code change -->
1. View the Customer Account Create
2. Apply changes
3. Enable the configuration setting (Stores > Settings > Configuration > Customer > Customer Configuration > Create New Account Options > Show Address Fields
4. View the Customer Account Create and fill in the form


### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

lewisvoncken committed:
[TASK] Added configuration option to show Address Fields on Registration Page

lewisvoncken committed:
[TASK] Added check if Address Attribute company and/or fax are allowed in registration form

vrann commented on app/code/Magento/Customer/Block/Form/Register.php:
```diff
@@ -220,4 +244,35 @@ public function getRequiredCharacterClassesNumber()
+     *
+     * @return bool
+     */
+    public function getShowAddressFields()
```
where is this method used?

lewisvoncken replied:
@vrann this method is used in the template it was already there but not in use

vrann commented on app/code/Magento/Customer/Block/Form/Register.php:
```diff
@@ -220,4 +244,35 @@ public function getRequiredCharacterClassesNumber()
+        $entityType = AddressMetadataInterface::ENTITY_TYPE_ADDRESS;
+        $attributeId = $this->_eavAttribute->getIdByCode( $entityType, $attributeCode);
+        $bind = ['attribute_id' => $attributeId, 'form_code' => $formCode];
+        $select = $this->_resource->getConnection()->select()->from(
```
It is not a recommended approach having SQL queries in the Blocks. Ideally they should be placed in resource models.

lewisvoncken committed:
[TASK] Use method of Resource Model getUsedInForms instate of query in Block

lewisvoncken committed:
[TASK] Renamed Block Function getShowAddressFields to isShowAddressFields

 - Added Backwards Compatibilty by adding getShowAddressFields to the template
 - getShowAddressFields can still be used in Third Party Templates to display Address Fields on Registration

lewisvoncken commented:
@vrann I have just made changes based on the feedback

lewisvoncken commented:
@vrann do you have any questions ?

magento-engcom-team commented:
@lewisvoncken thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

Commit:
MAGETWO-87652: Experius patch registration address #11655

 - fixed backward compatibility

okorshenko commented:
Hi @lewisvoncken 
Here is what we got from QA department:
QA/QC Report
## Steps to reproduce

1. Enable the configuration setting (Stores > Settings > Configuration > Customer > Customer Configuration > Create New Account Options > Show Address Fields
2. Flush cache
2. Got to store front. Customer Account Create page and fill in the form

✅ Expected Result: Text field State/Province present.

 ❌ Actual result: Not present text field State/Province
![stateprovince](https://user-images.githubusercontent.com/7325075/36760628-47ed4f6c-1be1-11e8-90e6-7df7f2e71fd8.png)


 Note: I can create customer with blank required State/Province field.

okorshenko commented:
Hi @lewisvoncken 
You can see this behavior here:
http://pr-11655.engcom.dev.magento.com/
(Admin access: admin: 123123q)

okorshenko commented:
Hi @lewisvoncken please let us know if you would like proceed with this PR

okorshenko commented:
Hi @lewisvoncken  We are closing this PR due to inactivity. Feel free to reopen it once ready
